### PR TITLE
(test) O3-2113 : Setup playwright and add login e2e test

### PR DESCRIPTION
**Pull request by RandilaP**
## Requirements
- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.

#### For changes to apps
- [x]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [x] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary
The Goal of this PR is to setup the core repository for E2E testing with playwright and add login and Logout E2E test

## Screenshots
image

## Related Issue

https://issues.openmrs.org/browse/O3-2113

### CODE REVIEW

**Comment by RandilaP** [comment]
@Piumal1999 @jayasanka-sack Could you please review this

**Comment by RandilaP** [comment]
@anjula-sack @Piumal1999 @jayasanka-sack Could you please run the workflows

**Comment by RandilaP** [comment]
@anjula-sack @Piumal1999 @jayasanka-sack Could you please approve approve the workflow

**Review by anjula-sack — CHANGES_REQUESTED** [review]
@RandilaP This should be ignored

**Comment by RandilaP** [comment]
@anjula-sack Could you please approve the workflow

File: e2e/storageState.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (anjula-sack) [inline e2e/storageState.json:1]: Remove this as well

**Comment by RandilaP** [comment]
@jayasanka-sack

**Comment by RandilaP** [comment]
@anjula-sack @jayasanka-sack @Piumal1999

**Comment by RandilaP** [comment]
@anjula-sack @jayasanka-sack @Piumal1999

**Comment by RandilaP** [comment]
@anjula-sack @jayasanka-sack @Piumal1999 Why is these report bundle and check documentations are failing

**Comment by RandilaP** [comment]
@anjula-sack @Piumal1999 @jayasanka-sack

**Comment by RandilaP** [comment]
@anjula-sack

**Comment by RandilaP** [comment]
@anjula-sack @jayasanka-sack @Piumal1999 The documentation is failing because of a link that has 404 error. It isn't my change. What should I do

File: e2e/README.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+```sh
+yarn start --sources 'packages/esm-*-app/'
+```
+Then, in a separate terminal, run:
+
+```sh
+yarn test-e2e --headed
+```
+
+By default, the test suite will run against the `http://localhost:8080`.
```
Comment (anjula-sack) [inline e2e/README.md:21]:
```suggestion
By default, the test suite will run against the "http://localhost:8080".
```

File: e2e/README.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+```sh
+yarn start --sources 'packages/esm-*-app/'
+```
+Then, in a separate terminal, run:
+
+```sh
+yarn test-e2e --headed
+```
+
+By default, the test suite will run against the "http://localhost:8080".
```
Comment (anjula-sack) [inline e2e/README.md:21]:
```suggestion
<!-- markdown-link-check-disable-next-line -->
By default, the test suite will run against the `http://localhost:8080`.
```

**Comment by anjula-sack** [comment]
@RandilaP Go to this file https://raw.githubusercontent.com/openmrs/openmrs-esm-core/main/docs/main/upgrade_3_to_4.md
 and replace the link `https://jestjs.io/docs/28.x/migration-guide` with this link `https://jestjs.io/blog/2022/04/25/jest-28`

**Comment by RandilaP** [comment]
@anjula-sack @jayasanka-sack @Piumal1999

File: e2e/fixtures/index.ts
Comment (anjula-sack) [inline e2e/fixtures/index.ts:1]: remove this empty file

File: e2e/commands/index.ts
Comment (anjula-sack) [inline e2e/commands/index.ts:1]: This too

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Comment by RandilaP** [comment]
@anjula-sack

File: e2e/specs/login.spec.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      .locator("#username")
+      .fill(`${process.env.E2E_USER_ADMIN_USERNAME}`);
+    await page.getByText("Continue").click();
+  });
+
+  await test.step("And I enter the password", async () => {
+    await page
+      .locator("#password")
+      .fill(`${process.env.E2E_USER_ADMIN_PASSWORD}`);
+    await page.getByText("Log in").click();
```
Comment (anjula-sack) [inline e2e/specs/login.spec.ts:23]: This is a step. @RandilaP

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Comment by RandilaP** [comment]
@anjula-sack

File: .github/workflows/e2e.yml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: Cache dependencies
+        id: cache
+        uses: actions/cache@v3
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
```
Comment (anjula-sack) [inline .github/workflows/e2e.yml:34]: This is not required @RandilaP 
```suggestion
```

File: .github/workflows/e2e.yml [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
```
Comment (anjula-sack) [inline .github/workflows/e2e.yml:79]:
```suggestion
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Comment by RandilaP** [comment]
@anjula-sack

**Comment by anjula-sack** [comment]
@RandilaP Can you undo the change?

**Comment by RandilaP** [comment]
@anjula-sack

**Comment by anjula-sack** [comment]
@RandilaP Add cache dependency back

**Comment by anjula-sack** [comment]
Replace this link as well https://zeroheight.com/23a080e38/p/880723--introduction

**Comment by RandilaP** [comment]
> Replace this link as well https://zeroheight.com/23a080e38/p/880723--introduction

@anjula-sack With which link

**Comment by anjula-sack** [comment]
> > Replace this link as well https://zeroheight.com/23a080e38/p/880723--introduction
> 
> @anjula-sack With which link

Just remove it @RandilaP

**Comment by RandilaP** [comment]
@anjula-sack What link should I replace with that failing link

**Comment by anjula-sack** [comment]
> @anjula-sack What link should I replace with that failing link

Just remove it

**Comment by RandilaP** [comment]
@anjula-sack

**Review by jayasanka-sack — APPROVED** [review]
Thank you @RandilaP! The link check job has failed because Jira is down. Therefore, I will merge the PR.

**Pull request merged**